### PR TITLE
feat: Add azure pipelines extension

### PR DESCRIPTION
**Pull request by andrei-bitca-dc**
Screenshot 2020-10-09 at 15 05 42

### CODE REVIEW

File: azure-devops-extension/run-automator-build-task/package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    "eslint-plugin-import": "^2.22.0",
+    "eslint-plugin-jest": "^24.0.2",
+    "eslint-plugin-json-format": "^2.0.1",
+    "eslint-plugin-no-loops": "^0.3.0",
+    "eslint-plugin-node": "^11.1.0",
+    "eslint-plugin-promise": "^4.2.1",
+    "eslint-plugin-sonarjs": "^0.5.0",
+    "jest": "^26.5.2",
+    "prettier": "^2.1.2",
+    "tfx-cli": "^0.8.3",
```
Comment (izy-izzy) [inline azure-devops-extension/run-automator-build-task/package.json:32]: Do we need this here if it is in the package.json in parent directory?

File: azure-devops-extension/run-automator-build-task/src/automator-sdk-client.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import { AutomatorSDKClient } from "@sdk/client";
```
Comment (izy-izzy) [inline azure-devops-extension/run-automator-build-task/src/automator-sdk-client.ts:1]: Do we need this file if only thing that it does is to create an instance?

File: azure-devops-extension/vss-extension.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+  "manifestVersion": 1,
+  "id": "automator-tools",
+  "name": "Automator Tools",
```
Comment (izy-izzy) [inline azure-devops-extension/vss-extension.json:4]: Is it really Automator Tools? Should it be named like 'Automator Run Build Script' ?

**Review by izy-izzy — COMMENTED** [review]
Looks good. Should we do also tests for this or we are going to leave it as it is ?

File: azure-devops-extension/run-automator-build-task/package.json (hunk 1 above)
Comment (andrei-bitca-dc) [reply azure-devops-extension/run-automator-build-task/package.json:32]: Should be remove indeed.

File: azure-devops-extension/run-automator-build-task/src/automator-sdk-client.ts (hunk 2 above)
Comment (andrei-bitca-dc) [reply azure-devops-extension/run-automator-build-task/src/automator-sdk-client.ts:1]: It's using the singleton module pattern. We can keep it, or we can remove it :).

File: azure-devops-extension/vss-extension.json (hunk 3 above)
Comment (andrei-bitca-dc) [reply azure-devops-extension/vss-extension.json:4]: Think of an extension like a Task Suite. `Automator Tools` extension contains multiple tasks, one of which is: `Automator Run Build Script`. In our case this is the only task that it contains 😄.

**Comment by andrei-bitca-dc** [comment]
> Looks good. Should we do also tests for this or we are going to leave it as it is ?

We can add tests, by mocking graph-api and tools-api calls.

File: azure-devops-extension/run-automator-build-task/src/sdk/tools-api/client.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        ciBuildId,
+      },
+    });
+    return triggerResponse.data.buildId;
+  }
+
+  public async poll(token: string, buildId: string, currentRunTime = 0): Promise<void> {
+    try {
+      const didTestsPass = await this.getResults(token, buildId);
+      console.log(`DeepCrawl Tests ${didTestsPass ? "Passed" : "Failed"}`);
```
Comment (AndKiel) [inline azure-devops-extension/run-automator-build-task/src/sdk/tools-api/client.ts:69]: nit: I would rather not use console anywhere. It's part of `dom` which isn't present in tsconfig `lib` value at the moment. It's nicer to use dedicated logger like `pino` or any other that supports standard output if the logging to console is needed.

**Review by AndKiel — APPROVED** [review]
LGTM

File: azure-devops-extension/run-automator-build-task/package.json [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    "nock": "^13.0.4",
+    "prettier": "^2.1.2",
+    "ts-jest": "^26.4.1",
+    "ttypescript": "^1.5.12",
+    "typescript": "^4.0.3"
+  },
+  "dependencies": {
+    "@apollo/client": "^3.2.2",
+    "@types/node-fetch": "^2.5.7",
+    "@types/q": "^1.5.4",
```
[the remark this answers is not in the document]
Comment (andrei-bitca-dc) [reply azure-devops-extension/run-automator-build-task/package.json:50]: weird they ended up there, don't remember installing them

File: azure-devops-extension/run-automator-build-task/.eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
+module.exports = {
+  parser: "@typescript-eslint/parser",
+  parserOptions: {
+    createDefaultProgram: true,
+    ecmaVersion: 2018,
```
[the remark this answers is not in the document]
Comment (andrei-bitca-dc) [reply azure-devops-extension/run-automator-build-task/.eslintrc.js:5]: yeap, mismatch

File: azure-devops-extension/run-automator-build-task/src/sdk/tools-api/client.ts (hunk 4 above)
Comment (andrei-bitca-dc) [reply azure-devops-extension/run-automator-build-task/src/sdk/tools-api/client.ts]: done, nice catch :)

**Review by AndKiel — DISMISSED** [review]
LGTM

**Review by AndKiel — DISMISSED** [review]

**Review by izy-izzy — APPROVED** [review]

**Pull request merged**